File: find-xor-sum-of-all-pairs-bitwise-and.cc
// Find XOR Sum of All Pairs Bitwise AND
class Solution {
public:
  int getXORSum(vector<int>& a, vector<int>& b) {
    return accumulate(a.begin(), a.end(), 0, bit_xor()) &
      accumulate(b.begin(), b.end(), 0, bit_xor());
  }
};

File: find-xor-sum-of-all-pairs-bitwise-and.rs
impl Solution {
    pub fn get_xor_sum(arr1: Vec<i32>, arr2: Vec<i32>) -> i32 {
        use std::ops::BitXor;
        arr1.iter().fold(0, BitXor::bitxor) & arr2.iter().fold(0, BitXor::bitxor)
    }
}
